src/EditorFeatures/Core.Wpf/Workspaces/WpfTextBufferVisibilityTracker.cs

```diff
@@ -110,14 +112,38 @@ public void UnregisterForVisibilityChanges(ITextBuffer subjectBuffer, Action cal
             }
         }
 
+        public TestAccessor GetTestAccessor()
+            => new TestAccessor(this);
+
+        public struct TestAccessor
+        {
+            private readonly WpfTextBufferVisibilityTracker _visibilityTracker;
+
+            public TestAccessor(WpfTextBufferVisibilityTracker visibilityTracker)
+            {
+                _visibilityTracker = visibilityTracker;
+            }
+
+            public void TriggerCallbacks(ITextBuffer subjectBuffer)
+            {
+                var data = _visibilityTracker._subjectBufferToCallbacks[subjectBuffer];
+                data.TriggerCallbacks();
+            }
+        }
+
         private sealed class VisibleTrackerData : IDisposable
         {
-            public readonly HashSet<Action> Callbacks = new();
             public readonly HashSet<ITextView> TextViews = new();
 
             private readonly WpfTextBufferVisibilityTracker _tracker;
             private readonly ITextBuffer _subjectBuffer;
 
+            /// <summary>
+            /// The callbacks that want to be notified when our <see cref="TextViews"/> change visibility.  Stored as an
+            /// <see cref="ImmutableHashSet{T}"/> so we can enumerate it safely without it changing underneath us.
+            /// </summary>
+            public ImmutableHashSet<Action> Callbacks { get; private set; } = ImmutableHashSet<Action>.Empty;
```

Comment:
❔ Do we care about order here? Normal events always and `HashSet<T>` often preserves "registration order is callback order", but `ImmutableHashSet<T>` will not.

Reply:
Good question. I hadn't ever thought it was relevant, but I'm ok adding that constraint.

Reply:
For now, i'm ok with this as is.  We coudl def make this an array in teh future though if we think it's appropriate.